### DISCUSSION TITLE
ERROR    request license token faild (3) - after rebooting a registered VM using the group license

### DISCUSSION THREAD

**Discussion by goraje** [dashingsoft/pyarmor#1327] [Q&A]
Hello, I've noticed that after rebooting a virtual machine following a successful registration of Pyarmor's group license I get the following error when trying to execute `pyarmor gen`:
```
ERROR    request license token faild (3)
```
and when I try to run `pyarmor reg pyarmor-device-regfile-<xxxx>.<device_id>.zip` to validate the registration I get:

```
ERROR    this regfile is not for this device
```

The VM was created on a M1 Mac using UTM (`qemu` is the actual hypervisor underneath). I have made sure that both the MAC address and UUIDs listed through `blkid` are still the same before and after the reboot, however even though they remain unchanged the VM is still not being recognised.

At our company the security regulations require us to use build environments that utilize ephemeral virtual machines that are recreated for each CI run. In that context what would be the proper approach to working with the group license taking into consideration the aformentioned issue?

**Comment by jondy** [comment]
Also keep hostname is same.

Generally if the image is stable, the device registration file is same.

If using docker, pyarmor 8.2.8 introduces new feature to support unlimited docker containers, refer to
https://pyarmor.readthedocs.io/en/v8.2.8/how-to/register.html#run-unlimited-dockers-in-offline-device
